### DISCUSSION TITLE
Extract annotations and regular text separately?

### DISCUSSION THREAD

**Discussion by hellpanderrr** [pymupdf/PyMuPDF#1726] [Looking for help] [answered]
Hi.
I am wondering if it is possible to extract text items using `Page.get_text()` or something similar, while excluding text from annotations?
I got some PDFs where annotations were used to cover up the old values with revised ones, so I need a way to differentiate them. Currently -- if I understand it correctly -- the annotations' content gets duplicated, i.e. output of `[i.info['content'] for i in page.annots()]` is contained within `page.get_text()`.

**Comment by JorjMcKie** [comment, marked as answer]
There is a rather hacky way
1. Extract annotation text
2. Remove annotations
3. Extract remaining page text

To remove annotations, 
```python
# do either this:
a = page.first_annot
while a:
    a = page.delete_annot(a)

# or this:
doc.xref_set_key(page.xref, "Annots", "null")
```
This will not change the document permanently if you don't save. Doing `doc.close();doc=fitz.open(doc.name)` will recycle the document in original state.
